openmetadata-ui/src/main/resources/ui/playwright/e2e/Pages/DataInsight.spec.ts

```diff
@@ -0,0 +1,274 @@
+  await page.getByTestId('displayName').fill(data.displayName);
+  await page.getByTestId('metricType').click();
+  await page.click(`.ant-select-dropdown [title="${data.metricType}"]`);
+  await page.locator('.ant-slider-mark-text', { hasText: '100%' }).click();
+
+  await page.getByTestId('start-date').click();
+  await page.getByTestId('start-date').fill(startDate);
+  await page.getByTestId('start-date').press('Enter');
+  await page.getByTestId('end-date').click();
+  await page.getByTestId('end-date').fill(endDate);
+  await page.getByTestId('end-date').press('Enter');
+
+  await page.locator(descriptionBox).fill('Playwright KPI test description');
+
+  await page.getByTestId('submit-btn').click();
+  await page.waitForURL('**/data-insights/kpi');
+};
+
+test.describe.configure({ mode: 'serial' });
+
+test.describe('Data Insight Page', () => {
+  test.beforeAll(async ({ browser }) => {
+    const { apiContext } = await createNewPage(browser);
+
+    await deleteKpiRequest(apiContext);
+  });
+
+  test.beforeEach('Visit Data Insight Page', async ({ page }) => {
+    await redirectToHomePage(page);
+    await sidebarClick(page, SidebarItem.DATA_INSIGHT);
+  });
+
+  test('Create description and owner KPI', async ({ page }) => {
+    // await page.getByTestId('loader').waitFor({ state: 'detached' });
+
+    await page.getByRole('menuitem', { name: 'KPIs' }).click();
+
+    for (const data of KPI_DATA) {
+      await page.getByTestId('add-kpi-btn').click();
+
+      await addKpi(page, data);
+    }
+  });
+
+  test('Run DataInsight Application', async ({ page }) => {
+    await settingClick(page, GlobalSettingOptions.APPLICATIONS);
+
+    await page.click(
+      '[data-testid="data-insights-application-card"] [data-testid="config-btn"]'
+    );
+    await page.waitForResponse('**/api/v1/apps/name/DataInsightsApplication?*');
+
+    await page.click('[data-testid="run-now-button"]');
+    await page.waitForResponse(
+      '**/api/v1/apps/trigger/DataInsightsApplication'
+    );
+
+    const { apiContext } = await getApiContext(page);
+
+    await expect
+      .poll(
+        async () => {
+          const response = await apiContext
+            .get(
+              '/api/v1/apps/name/DataInsightsApplication/status?offset=0&limit=1'
+            )
+            .then((res) => res.json());
+
+          return response.data[0].status;
+        },
+        {
+          // Custom expect message for reporting, optional.
+          message: 'Wait for the pipeline to be successful',
+          timeout: 60_000,
+          intervals: [5_000, 10_000],
+        }
+      )
+      .toBe('success');
+
+    // Verify KPI
+
+    await sidebarClick(page, SidebarItem.DATA_INSIGHT);
+
+    await page.waitForSelector('[data-testid="search-dropdown-Team"]');
+    await page.waitForSelector('[data-testid="search-dropdown-Tier"]');
+    await page.waitForSelector('[data-testid="summary-card"]');
+    await page.waitForSelector('[data-testid="kpi-card"]');
```

Comment:
Here we are just waiting for the element, can we also add an assersion for the same?